src/main/java/de/hpi/bpt/argos/storage/DatabaseAccessImpl.java

```diff
@@ -144,9 +146,51 @@ public boolean deleteArtifacts(PersistenceArtifact... artifacts) {
 	 * {@inheritDoc}
 	 */
 	@Override
-	public <R, Q> R getArtifacts(Session session, Query<Q> query, Transaction transaction, Callable<R> getValue, R defaultValue) {
+	public <ResultType, QueryType> ResultType getArtifacts(Session session,
+														   Query<QueryType> query,
+														   Transaction transaction,
+														   Callable<ResultType> getValue,
+														   ResultType defaultValue) {
+
+		ObjectWrapper<ResultType> result = new ObjectWrapper<>();
+		WatchImpl.measure("get artifacts",
+				() -> result.set(executeGetArtifacts(session, query, transaction, getValue, defaultValue)));
+
+		return result.get();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public <ResultType, ResultImplType extends ResultType> List<ResultType> getArtifactsById(Session session,
+																							 Class<ResultImplType> resultTypeClass,
+																							 Long... ids) {
+
+		ObjectWrapper<List<ResultType>> result = new ObjectWrapper<>();
+		WatchImpl.measure("get artifacts by id",
+				() -> result.set(executeGetArtifactsById(session, resultTypeClass, ids)));
+		return result.get();
+	}
+
+	/**
+	 * This method actually executes the getArtifact-method
+	 * @param session - the database session, which must be open
+	 * @param query - the query to execute and to retrieve the results from
+	 * @param transaction - the current transaction
+	 * @param getValue - the function to get the results from the query
+	 * @param defaultValue - a fall back default value in case anything went wrong or no entities were found
+	 * @param <ResultType> - the result type
+	 * @param <QueryType> - the query type
+	 * @return - an object of the result value type
+	 */
+	private <ResultType, QueryType> ResultType executeGetArtifacts(Session session,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)